docs/style-guide.rst

```diff
@@ -51,90 +50,558 @@ into the making of Hy.
+               (= n 0.0)
+               "zero"
+               (> n 0.0)
+               "positive"
+               "not a number"))
+
+    ;; This is also acceptable.
+    (print (if (< n 0.0) "negative"
+               (= n 0.0) "zero"
+               (> n 0.0) (do (do-foo)  ; Group started this line, so didn't break.
+                             (do-bar)
+                             "positive")
+               "not a number"))  ; :else is implied for the last one.
+
+    ;; Bad.
+    (print (if (< n 0.0) "negative"
+               (= n 0.0) "zero"
+               (and (even? n)
+                    (> n 0.0)) "even-positive"  ; Group not started this line! Should break on "))"
+               (> n 0.0) "positive"
+               "not a number"))
+
+    ;; Worse.
+    (print (if (< n 0.0) "negative"
+               (= n 0.0) "zero"
+               (and (even? n)
+                    (> n 0.0)) (do (do-foo)  ; Group not started this line.
+                                   (do-bar)
+                                   "even-positive")
+               (> n 0.0) "positive"
+               "not a number"))
+
+    ;; Good. Blank line separates groups.
+    (print (if (< n 0.0) "negative"
+
+               (= n 0.0) "zero"
+
+               (and (even? n)
+                    (> n 0.0))
+               (do (do-foo)
+                   (do-bar)
+                    "even-positive")
+
+               (> n 0.0) "positive"
+
+               "not a number"))
+
+    ;; Not so good, groups are not separated consistently.
+    (print (if (< n 0.0) "negative"
+               (= n 0.0) "zero"
+
+               (> n 0.0)
+               (do (do-foo)
+                   "positive")
+
+               "not a number"))
+
+    ;; Acceptable. All groups are separated the same way, with a blank like.
+    (print (if (< n 0.0) "negative"
+
+               (= n 0.0) "zero"
+
+               (> n 0.0)
+               (do (do-foo)
+                   "positive")
+
+               "not a number"))
+
+    (defn fib [n]  ; saw a "]", newline.
+      (if (<= n 2)  ; OK to break here. Since there's only one pair, we don't have to separate them.
+          n
+        (+ (fib (- n 1))  ; non-special indent is another whitespace separation technique.
+           (fib (- n 2)))))
 
-    ;; Hysterically ridiculous
-    (defn fib [n]
-      (if (<= n 2)
-        n
-        (+ (fib (- n 1)) (fib (- n 2)))
-      )
-    )  ; GAH, BURN IT WITH FIRE
+Comments
+--------
 
+Prefer docstrings to comments where applicable--in ``defn``, ``defclass``, and at the top of the module.
 
-+ Inline comments shall be two spaces from the end of the code; they
-  must always have a space between the comment character and the start
-  of the comment. Also, try to not comment the obvious.
+The ``(comment)`` macro is still subject to the three laws.
+If you're tempted to violate them, consider discarding a string instead with ``#_``.
+
+Semicolon comments shall start with some number of semicolons
```

Comment:
Maybe mention that semicolon count guidance is below?

Reply:
Too long before the example? @ekaschalk was suggesting we ditch the headers altogether, that would make this section shorter. Then we'd only have the `;` and `;;` styles.